presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java

```diff
@@ -114,7 +100,7 @@ public OrcBatchPageSource(
 
         this.stats = requireNonNull(stats, "stats is null");
         this.runtimeStats = requireNonNull(runtimeStats, "runtimeStats is null");
-        this.isRowPositionList = requireNonNull(isRowPositionList, "isRowPositionList is null");
+        this.isRowNumberList = Booleans.toArray(requireNonNull(isRowNumberList, "isRowNumberList is null"));
```

Comment:
Please add a size check that isRowNumbers.len >= columns.size()

Reply:
done